demos/imagenet_classification_demo/grid_mat.hpp

```diff
@@ -0,0 +1,176 @@
+        if (isFpsTest) {
+            cv::putText(outImg,
+                        testMessage,
+                        cv::Point(outImg.cols - testMessageSize.width - textPadding,
+                                  (textSize.height + textPadding) * 2),
+                        cv::FONT_HERSHEY_PLAIN, fontScale, cv::Scalar(50, 50, 255), thickness);
+        }
+    }
+
+    cv::Mat getMat(std::vector<long long> &endTimes, bool showLabels) {
+        while (!updateList.empty()) {
+            cv::Mat frame = std::get<0>(updateList.front());
+            std::string predictedLabel = std::get<1>(updateList.front());
+            cv::Scalar textColor = (std::get<2>(updateList.front())
+                                                ? cv::Scalar(75, 255, 75) : cv::Scalar(50, 50, 255));
+            updateList.pop_front();
+
+            if (updateList.empty()) {
+                if (prevImgs.size() >= FLAGS_b && points.size() > 1) {
+                    int prevSourceID = currSourceID - FLAGS_b;
+                    if (prevSourceID < 0) {
+                        prevSourceID += points.size();
+                    }
+                    cv::resize(prevImgs.front(),
+                               outImg(cv::Rect(points[prevSourceID], cellSize)),
+                               cellSize);
+                    prevImgs.pop();
+                }
+                
+                cv::resize(frame, frame, cellSize);
+
+                int thickness = cellSize.width / 20;
+                int baseline = 0;
+
+                if (showLabels) {
+                    cv::Size textSize = cv::getTextSize(predictedLabel, cv::FONT_HERSHEY_PLAIN, 1, 2, &baseline);
+                    double fontScale = static_cast<double>(cellSize.width - 2*thickness) / textSize.width;
+                    cv::putText(frame,
+                                predictedLabel,
+                                cv::Point(thickness, cellSize.height - thickness - textSize.height),
+                                cv::FONT_HERSHEY_PLAIN, fontScale, textColor, 2);
+                }
+                prevImgs.push(frame);
+
+                cv::Mat tmpFrame;
+                frame.copyTo(tmpFrame);
+                cv::rectangle(tmpFrame, cv::Point(), cv::Point(frame.cols, frame.rows),
+                              cv::Scalar(255, 50, 50), thickness);
+                cv::resize(tmpFrame,
+                           outImg(cv::Rect(points[currSourceID], cellSize)),
+                           cellSize);
+                
+
+                if (currSourceID == points.size() - 1) {
+                    currSourceID = 0;
+                } else {
+                    currSourceID++;
+                }
+            }
+
+            endTimes.push_back(cv::getTickCount());
+        }
+
+        return outImg;
+    }
+
+private:
+    cv::Mat outImg;
+    std::queue<cv::Mat> prevImgs;
+    std::list<std::tuple<cv::Mat, std::string, bool>> updateList;
+    cv::Size size;
```

Comment:
`size` is never modified. That allows to make it public const and remove `getSize()` method.

Reply:
Actually this field is never used except for in a constructor, thus it can be just a local var in the constructor.